frame/staking/src/pallet/mod.rs

```diff
@@ -559,6 +651,184 @@ pub mod pallet {
 	#[pallet::getter(fn current_planned_session)]
 	pub type CurrentPlannedSession<T> = StorageValue<_, SessionIndex, ValueQuery>;
 
+	/// Wrapper struct for Era related information. It is not a pure encapsulation as these storage
+	/// items can be accessed directly but nevertheless, its recommended to use `EraInfo` where we
+	/// can and add more functions to it as needed.
+	pub(crate) struct EraInfo<T>(sp_std::marker::PhantomData<T>);
+	impl<T: Config> EraInfo<T> {
+		/// Temporary function which looks at both (1) passed param `T::StakingLedger` for legacy
+		/// non-paged rewards, and (2) `T::ClaimedRewards` for paged rewards. This function can be
+		/// removed once `$HistoryDepth` eras have passed and none of the older non-paged rewards
+		/// are relevant/claimable.
+		// Refer tracker issue for cleanup: #13034
+		pub(crate) fn is_rewards_claimed_temp(
+			era: EraIndex,
+			ledger: &StakingLedger<T>,
+			validator: &T::AccountId,
+			page: PageIndex,
+		) -> bool {
+			ledger.legacy_claimed_rewards.binary_search(&era).is_ok() ||
+				Self::is_rewards_claimed(era, validator, page)
+		}
+
+		/// Check if the rewards for the given era and page index have been claimed.
+		///
+		/// This is only used for paged rewards. Once older non-paged rewards are no longer
+		/// relevant, `is_rewards_claimed_temp` can be removed and this function can be made public.
+		fn is_rewards_claimed(era: EraIndex, validator: &T::AccountId, page: PageIndex) -> bool {
+			ClaimedRewards::<T>::get(era, validator).binary_search(&page).is_ok()
+		}
+
+		/// Get exposure info for a validator at a given era and page.
+		///
+		/// This builds a paged exposure from `ExposureOverview` and `ExposurePage` of the
+		/// validator. For older non-paged exposure, it returns the clipped exposure directly.
+		pub(crate) fn get_validator_exposure(
+			era: EraIndex,
+			validator: &T::AccountId,
+			page: PageIndex,
+		) -> Option<ExposureExt<T::AccountId, BalanceOf<T>>> {
+			return match <ErasStakersPaged<T>>::get(era, (validator, page)) {
+				// return clipped exposure if page zero and paged exposure does not exist
+				None if page == 0 =>
+					Some(ExposureExt::from_clipped(<ErasStakersClipped<T>>::get(era, validator))),
```

Comment:
This is kinda janky because `ErasStaskersClipped` is `ValueQuery`. Ideally, we should have chekced if `ErasStakersClipped` exists, and only then return `Some(_)`.

Currently, if we call this function for an era that is wrong (in the future) it will return `Some(_)`. Although, since the `ExposureExt` is empty, nothing should break. 

Something to double check.

Reply:
I know 😬 . I scratched my head a lot around one test because of this. For older eras, logically nothing changes since even earlier we were returning an empty exposure and calculating rewards using it. But in a real world scenario, this will never get called since `EraRewards` for that validator would be empty too and code will break before we try to read exposure. 

But I at least wanted to improve on earlier code and return `None` when no exposure exists.